models/maespa/R/met2model.MAESPA.R

```diff
@@ -0,0 +1,265 @@
+      #if(!is.numeric(RHperc)==TRUE){print("Percent Relative Humidity(RH%) 
+      # will be calculated from VPD, VMFD, or TAIR by MAESPA")}
+      #mmol mol-1   vapour pressure mole fraction deficit   Calculated from RH or VPD, and PRESS
+      #if(!is.numeric(VMFD)== TRUE){print("Vapor pressure mole fraction deficit(VMFD) will be calulated from RH or VPD an PRESS by MAESPA")}
+      ####mm   Precipitation.   Assume zero precipitation.
+      #PPT <- units are fine (kg m-2 = mm). need to convert to hourly though
+      ####m s-1   wind speed above the canopy   Assumed = DEFWIND (defined in maestcom.f90)
+      #WIND <- units fine. need to convert to hourly
+      ####Pa   atmospheric pressure   Constant value from ENVIRON namelist used.
+      #PRESS <- units are fine. need to convert to houly
+      ####Same as SWMAX   Soil water content.   Not used.
+      #if(!is.numeric(SW)==TRUE){print("Soil Water potential will not be used y MAESPA")}
+      ####ppm. atmospheric CO2 concentration. Constant from Enviiron namelist used instead
+      if(!is.numeric(CA)==TRUE){print("Atmospheric CO2 concentration will be set to constant value set in ENVIRON namelist ")
+                                rm(CA)
+                                defaultCO2= 400 #400 is estimation of atmospheric CO2 in ppm)
+      }else{defaultCO2= 400} #400 is estimation of atmospheric CO2 in ppm))
+      
+      nc_close(nc)
+    } else {
+      print("Skipping to next year")
+      next
+    }
+    
+    ##build time variables (year, month, day of year)
+    skip = FALSE
+    nyr <- floor(length(sec)/86400/365*dt)
+    yr <- NULL
+    doy <- NULL
+    hr <- NULL
+    asec <- sec
+    for(y in year+1:nyr-1){
+      ytmp <- rep(y,365*86400/dt)
+      dtmp <- rep(1:365,each=86400/dt)
+      if(y %% 4 == 0){  ## is leap
+        ytmp <- rep(y,366*86400/dt)
+        dtmp <- rep(1:366,each=86400/dt)
+      }
+      if(is.null(yr)){
+        yr <- ytmp
+        doy <- dtmp
+        hr <- rep(NA,length(dtmp))
+      } else {
+        yr <- c(yr,ytmp)
+        doy <- c(doy,dtmp)
+        hr <- c(hr,rep(NA,length(dtmp)))
+      }
+      rng <- length(doy) - length(ytmp):1 + 1
+      if(!all(rng>=0)){
+        skip = TRUE
+        logger.warn(paste(year,"is not a complete year and will not be included"))
+        break
+      }
+      asec[rng] <- asec[rng] - asec[rng[1]]
+      hr[rng] <- (asec[rng] - (dtmp-1)*86400)/86400*24
+    }
+    if(length(yr) < length(sec)){
+      rng <- (length(yr)+1):length(sec)
+      if(!all(rng>=0)){
+        skip = TRUE
+        logger.warn(paste(year,"is not a complete year and will not be included"))
+        break
+      }
+      yr[rng] <- rep(y+1,length(rng))
+      doy[rng] <- rep(1:366,each=86400/dt)[1:length(rng)]
+      hr[rng] <- rep(seq(0,length=86400/dt,by=dt/86400*24),366)[1:length(rng)]
+    }
+    if(skip){
+      print("Skipping to next year")
+      next
+    }
+    tmp<-rbind(TAIR,PPT)
```

Comment:
After processing all the met variables listed above, why are you only writing out temp and precip?